docs/source/reference.adoc

```diff
@@ -360,3 +360,109 @@ because generating a new token is an expensive operation. Get the access token i
 curl -H "Content-Type: application/json" -X POST -d '{"refresh_token":"<REFRESH_TOKEN>"}' https://auth.openshift.io/api/token/refresh
 
 As a result, you will get a JSON with an access token which you can use in your test.
+
+== Token validation
+
+=== The JWT contents
+
+The token is likely to contain the following claims.
+
+==== Header
+
+The header contains the algorithm and the key ID "kid" used to sign the token.
+
+[source]
+{
+  "alg": "RS256",
+  "kid": "0lL0vXs9YRVqZMowyw8uNLR_yr0iFaozdQk9rzq2OVU",
+  "typ": "JWT"
+}
+
+==== Payload
+[source]
+ "acr": "0",
+  "allowed-origins": [
+    "https://auth.openshift.io",
+    "https://openshift.io"
+  ],
+  "approved": true,
+  "aud": "fabric8-online-platform",
+  "auth_time": 1534868165,
+  "azp": "fabric8-online-platform",
+  "email": "myusername@gmail.com",
+  "email_verified": false,
+  "exp": 1537460165,
+  "family_name": "Bose",
+  "given_name": "Shoubhik",
+  "iat": 1534868165,
+  "iss": "https://sso.openshift.io/auth/realms/fabric8",
+  "jti": "4ea61a85-f607-4491-b973-ef6314ae4c23",
+  "name": "Shoubhik Bose",
+  "preferred_username": "myusername",
+  "sub": "3383826c-51e4-401b-9ccd-b898f7e2397d",
+  "typ": "Bearer"
+}
+
+
+`sub` is the identity ID of the user.
+
+`jti` is the unique token identifier.
+
+=== Verify the signature
+This is necessary to verify that the JWT was issued by the Auth service and to ensure that the message wasn't changed along the way. 
+The signature has to be verified against the public key of the corresponding private key used to sign the JWT. 
+
+The list of public keys in use are available at https://auth.openshift.io/api/token/keys?format=pem 
+
+https://auth.openshift.io/api/token/keys will return keys in JSON Web Key (JWK) format. There are plenty of libraries which can be used to parse JWK keys
```

Comment:
maybe provide a link to the go jwt library that we use (github.com/dgrijalva/jwt-go)